crates/astria-sequencer/src/ibc/ics20_withdrawal.rs

```diff
@@ -50,20 +52,42 @@ use crate::{
     transaction::StateReadExt as _,
 };
 
-fn withdrawal_to_unchecked_ibc_packet(
+async fn create_ibc_packet_from_withdrawal<S: StateRead>(
     withdrawal: &action::Ics20Withdrawal,
-) -> IBCPacket<Unchecked> {
-    let packet_data = withdrawal.to_fungible_token_packet_data();
+    state: S,
+) -> anyhow::Result<IBCPacket<Unchecked>> {
+    let sender = if withdrawal.use_compat_address {
+        let ibc_compat_prefix = state.get_ibc_compat_prefix().await.context(
+            "need to construct bech32 compatible address for IBC communication but failed reading \
+             required prefix from state",
+        )?;
+        withdrawal
+            .return_address()
+            .to_prefix(&ibc_compat_prefix)
+            .context("failed to convert the address to the bech32 compatible prefix")?
+            .to_format::<Bech32>()
+            .to_string()
```

Comment:
same question here - why does the counterparty care about the return address (which is an astria address?)